src/XMakeBuildEngine/UnitTests/Evaluation/ProjectRootElementCache_Tests.cs

```diff
@@ -189,5 +189,61 @@ public void GetProjectRootElementChangedOnDisk2()
                 File.Delete(path);
             }
         }
+
+        /// <summary>
+        /// This test replicates the scenario in https://devdiv.visualstudio.com/DevDiv/_workitems?id=366077&_a=edit
+        /// Two different caches can interfere with each other when auto reloading from disk is turned on.
+        /// </summary>
+        [Fact]
+        public void GetProjectRootElementChangedOnDisk3()
+        {
+            string path = null;
+
+            try
+            {
+                var nonReloadingCache = new ProjectRootElementCache(autoReloadFromDisk: false);
+                var reloadingCache = new ProjectRootElementCache(autoReloadFromDisk: true);
+
+                path = FileUtilities.GetTemporaryFile();
+
+                var nonReloadingProject = ProjectRootElement.Create(path);
+                nonReloadingCache.AddEntry(nonReloadingProject);
+
+                nonReloadingProject.Save();
+
+                var previousReloadingProject = ProjectRootElement.Open(path, reloadingCache, true, null);
+                Assert.NotSame(nonReloadingProject, previousReloadingProject);
+
+                for (int i = 0; i < 30; i++)
```

Comment:
Yup, it was the first number after 3 that came to mind :)
Since there's no thread sleeps in the test it should go pretty fast.